### PR TITLE
427 Create Webpage Component

### LINKED ISSUE #427: Create Webpage Component
# Story: Create Webpage Component #443 

As a **user**, I want **a mock web environment**, so that **I can better understand the intent of Lab 9**.

Associated Epic: #409 

## Definition of Done

- [x] Acceptance criteria defined
- [x] Solution tasks specified
- [x] Feature branch created
- [ ] ~~Unit tests written (on API code only for now)~~
- [ ] ~~Solution passes full suite of unit tests~~
- [ ] Design documents updated if needed
- [x] Pull request created
- [ ] Code is reviewed
- [ ] Code is merged to `develop` branch
- [ ] ~~Tests pass on CI~~
- [ ] ~~Deployed to staging environment~~
- [x] Acceptance criteria is met

## Acceptance Criteria

- [x] Given **I am in the exercise**, when **I navigate to the simulation**, then **I can view the mock web environment**
- [x] Given **initial context**, when **event occurs**, then **some desired outcome**

## Solution Tasks

- [x] Implement changes to Figma's dev mode outputted web code and styles
- [x] Stub out a general component without any direct logic implementation
- [x] Structure the first solution task into separate components to better adhere to reusability.

### PR DESCRIPTION
**Pull request by heathermoses**
- [x] No discrepancies across browsers. Eg. chrome vs safari
- [x] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [x] CSS (https://jigsaw.w3.org/css-validator/)
- [x] Google sign-in is operational
- [x] Accessibility functions
- [x] Pages can scale without distorting page
- [x] No dead links
- [x] Navbar is consistent across the site
- [x] Pages are screen-reader accessible
- [x] Contrast meets standards for accessibility
- [x] Pages are keyboard accessible
- [x] Passes WAVE Evaluation Tool chrome extension

### CODE REVIEW

File: client/src/components/exercise/lab9/Main.js [hunk 1]
```diff
@@ -16,6 +17,7 @@ const Main = (props) => {
     <div className="bottomSpace">
       <GameStateContext.Provider>
         <Router className="app">
+          <Webpage path={"page"} />
```
Comment (heathermoses) [inline client/src/components/exercise/lab9/Main.js]: Will need to update once we have more of the exercise routing done

**Comment by heathermoses** [comment]
This PR is generally just to get the basic stubbed-out version of the webpage component working, later tasks will finalize styling

File: client/src/components/exercise/lab9/Main.js (hunk 1 above)
Comment (AndreasLc1103) [reply client/src/components/exercise/lab9/Main.js]: 👍🏼

**Comment by AndreasLc1103** [comment]
Just a quick thing but could you write doc strings on these display components, just so we can have them documented as we go, I probably can't approve it bc I worked on it so I would see if you can get more reviewers. From my perspective, it looks good. Awesome job @heathermoses ✅

File: client/src/components/exercise/lab9/components/Webpage.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import NewsletterForm from "./webpage-subcomponents/NewsletterForm";
+import WebpageNav from "./webpage-subcomponents/WebpageNav";
+import WebpageHeader from "./webpage-subcomponents/WebpageHeader";
+import WebpageSidebar from "./webpage-subcomponents/WebpageSidebar";
+
+const Webpage = () => {
+  return (
+    <div className="tw-bg-white tw-flex tw-flex-col">
+      <div className="tw-flex-col tw-bg-white tw-self-stretch tw-relative tw-flex tw-min-h-[1024px] tw-w-full tw-max-md:tw-max-w-full">
```
Comment (notjoncruz) [inline client/src/components/exercise/lab9/components/Webpage.js:19]: are we okay with defining a minimum heigh of 1024px? can this cause issues if the window size is smaller than that?

**Comment by heathermoses** [comment]
> Just a quick thing but could you write doc strings on these display components, just so we can have them documented as we go, I probably can't approve it bc I worked on it so I would see if you can get more reviewers. From my perspective, it looks good. Awesome job @heathermoses ✅

Good call, I just added docstrings to the main components!

**Review by notjoncruz — APPROVED** [review]
lgtm! the only major comment i have is whether we want to leave the hard-coded values in the style classnames.

**Pull request closed without merging**